pkg/prop/prop.go

```diff
@@ -16,6 +17,35 @@ type Media struct {
 	Codec
 }
 
+// Merge merges all the field values from o to p, except zero values.
+func (p *Media) Merge(o Media) {
+	rp := reflect.ValueOf(p).Elem()
+	ro := reflect.ValueOf(o)
+
+	// merge b fields to a recursively
+	var merge func(a, b reflect.Value)
+	merge = func(a, b reflect.Value) {
+		numFields := a.NumField()
+		for i := 0; i < numFields; i++ {
+			fieldA := a.Field(i)
+			fieldB := b.Field(i)
+
+			// if a is a struct, b is also a struct. Then,
+			// we recursively merge them
+			if fieldA.Kind() == reflect.Struct {
+				merge(fieldA, fieldB)
+				continue
+			}
+
+			if !fieldB.IsZero() {
```

Comment:
`IsZero` is available since 1.13.
```suggestion
			if !reflect.DeepEqual(reflect.Zero(fieldB), fieldB) {
```
would be a Go 1.12 compatible equivalent.

Reply:
Oops, I just saw your comment. This makes more readable. I'll update the PR

Reply:
I just tested it. It didn't seem to work... Some of the tests failed. `fieldB.Interface() != reflect.Zero(fieldB.Type()).Interface()` seems to be working better.

Reply:
Oops, I didn't tested that.
Current one seems nice.